### ISSUE TITLE
Made Bitcoin user aware if a deposit is not observe because of deposit fee

### ISSUE THREAD

**Issue by lumtis** [zeta-chain/node#3193]
**Describe the Issue**

A user should be aware if a deposit can't be processed because of depositFee

Eventually it is addressed in the toolkit https://github.com/zeta-chain/toolkit, prevent doing deposit less than the deposit fee

**Comment by ws4charlie** [comment]
So, in order to allow the users to track the status of a failed inbound, we need to record a status for each `inboundHash`. This status can be on-chain or off-chain. For successful ones, we already have CCTX status, for failed ones

If on-chain:
    the error will be written to Zetachain or Bitcoin.

If off-chain:
    the error will be written to some database/server/indexer and queries needs to be publicly available to users.

I think @fadeev could also share some insights from developer experience perspective.

**Comment by fadeev** [comment]
I understand the concerns @kingpinXD outlined, but having this on-chain is still preferable compared to an off-chain solution with a database.

Alternatively, we write a client-side function that can definitively tell if a specific tx has been processed or rejected due to deposit fees.

**Comment by kingpinXD** [comment]
I see your point; however, even if there are 9 individual DBSs, there might be inconsistency between them. Would that be an option for exploitation? 
I would assume this error message would only be used for debugging and showing an error message on explorer.  There is no interaction with the core at all .

**Comment by ws4charlie** [comment]
Yeah. The data inconsistency, lacking persistency and public access availability are the concerns.

**Comment by kingpinXD** [comment]
Yeah that's true 
I also think 1.31$, for now, is more than the fee paid by the observers to vote, even if we account for all the nine votes 
But 
- The price of zeta might go up 
- We can increase the gas-price (We were considering doing that slowly over time ) 
- The number of observers might increase, etc

I do think having the cctx is the best for user experience, but we should exhaust all other options before finalizing this

**Comment by fadeev** [comment]
If we can't do it on-chain, I still think having a bullet-proof function that returns whether an inbound has been processed or not is the way to go.

**Comment by kingpinXD** [comment]
@ws4charlie @lumtis  , I have been playing around with the zetatools; I think we can easily add an API to zetatools , which can return the required information without registering a CCTX for it. Would you guys consider trying that out ,and see if it solves all the current problems before trying to modify the core logic ?

**Comment by lumtis** [comment]
I think it's still better to move forward with the PR. Even if we can report the tx with a tool, this is still a tx supposed to be an inbound in theory since sent to the chain. The PR adds the notion of failed inbound observation that will be used further in the future (example: invalid receiver address with Bitcoin deposit v2)

**Comment by kingpinXD** [comment]
Okay, I just wanted to make sure we exhaust all options; 

The PR would also help if, in the future, we want to re-enable/ improve the accounting for tokens as well
